Desktop/analysis/analysisform.cpp

```diff
@@ -683,10 +686,22 @@ void AnalysisForm::blockValueChangeSignal(bool block, bool notifyOnceUnblocked)
 	else
 	{
 		_signalValueChangedBlocked--;
-		if (_signalValueChangedBlocked < 0)	_signalValueChangedBlocked = 0;
+		
+		if (_signalValueChangedBlocked < 0)	
+			_signalValueChangedBlocked = 0;
 
-		if (_signalValueChangedBlocked == 0 && notifyOnceUnblocked && _analysis)
-			_analysis->boundValueChangedHandler();
+		if (_signalValueChangedBlocked == 0)
+		{
+			if(notifyOnceUnblocked && _analysis)
+				_analysis->boundValueChangedHandler();
+		
+			while(_waitingRScripts.size() > 0)
```

Comment:
The _waitingRScripts must be run only if the notifyOnceUnblocked is true (maybe this variable can be renamed by 'runOnceUnblocked'). This notifyOnceUnblocked is false when reading a JASP file: the option must be set (and some options generate R script), but the analysis must not be run, it must stay in the state read from the JASP file.

Reply:
But the problem here is that the way we read it we miss some information and the `waiting rscript` is the one that encodes and checks it and returns this missing information. It doesn't need to run the analysis.

Reply:
And also, the state in the jasp file is changed but only augmented really